views/Grades/GradeView.js

```diff
@@ -155,16 +159,48 @@ function calculateExactGrades(grades, isClass) {
 }
 
 function GradeView({ route, navigation }) {
+  const appctx = useAppContext();
   const theme = useTheme();
   const { grade, allGrades } = route.params;
   const UIColors = GetUIColors();
+  
+  async function getName() {
+    var user = await appctx.dataprovider.getUser()
+    return user.name.split(' ').pop();
+  }
 
-  function shareGrade() {
-    Alert.alert(
-      'Partager la note',
-      "Le partage de la note n'est pas encore disponible.",
-      [{ text: 'OK' }]
-    );
+  async function shareGrade(grade, color) {
+    const baseURL = "https://getpapillon.xyz/note"
+    const comment = ["Une franche réussite","Qu'elle succès","Absolument fantastique","Épatant","Un travail exceptionnel","Remarquable","Impressionnant","Merveilleux","À couper le souffle","Brillant","Magique","Extraordinaire","Stupéfiant","Inoubliable","Exceptionnellement bien fait","Éblouissant","Génial","Sublime","Incroyable","Époustouflant","Travail consciencieux","Bonne performance","En progrès constant","Très réussi","Mérite une mention","Bien joué !","Excellent travail !","Bravo, très bien !","Très satisfaisant","Félicitations !","Travail exceptionnel",]
```

Comment:
Ce n'est pas plutôt "Quel succès" ?
```suggestion
    const comment = ["Une franche réussite","Quel succès","Absolument fantastique","Épatant","Un travail exceptionnel","Remarquable","Impressionnant","Merveilleux","À couper le souffle","Brillant","Magique","Extraordinaire","Stupéfiant","Inoubliable","Exceptionnellement bien fait","Éblouissant","Génial","Sublime","Incroyable","Époustouflant","Travail consciencieux","Bonne performance","En progrès constant","Très réussi","Mérite une mention","Bien joué !","Excellent travail !","Bravo, très bien !","Très satisfaisant","Félicitations !","Travail exceptionnel",]
```

Reply:
si ptdr

Reply:
ChatGPT... 💀